test/stringio/test_stringio.rb

```diff
@@ -978,6 +978,27 @@ def test_coderange_after_overwrite
     assert_predicate(s.string, :ascii_only?)
   end
 
+  if eval(%{ "test".frozen? && !"test".equal?("test") }) # Ruby 3.4+ chilled strings
```

Comment:
Why do we need to use `eval` here?

Reply:
Because there is `# frozen_string_literal: false` at the top of the file. 

An alternative would be to create another test file, or to remove it from this one.